### ISSUE TITLE
[Integration test] Allow to run integration test (with flutter test command) on profile mode

### ISSUE THREAD

**Issue by lhimo** [flutter/flutter#112959]
## Use case

On our CI pipeline we run the tests using `flutter test integration_test` command, and it works well since it has a nice reporting capability. The tests are though running very slowly as they run in debug mode.

## Proposal

We would like to be able to run integration tests on profile mode, while keeping the JUnit reporting functionality that is missing with `flutter drive` command.

**Comment by jonahwilliams** [comment]
Running the tests in profile mode would improve the runtime speed of the tests, but it would slow down compilation by many times more.

If you want to run tests in profile mode I would recommend using the flutter drive command instead, at least for now

**Comment by darshankawar** [comment]
@lhimo 
Check above recommendation and see if suits your use case.

**Comment by lhimo** [comment]
I agree, for a small set of tests it's better to run on debug. But if I have a large set it's more convenient to wait a bit longer building the app, as the tests will take way less time.

Flutter drive does not provide a test report that we need to export at the end of the pipeline.

**Comment by darshankawar** [comment]
Thanks for the feedback. Labeling it as a potential proposal.

**Comment by lhimo** [comment]
As a workaround, [this](https://gist.github.com/lhimo/553fb82704ef9bc8914f7046d7b3b593) script allows to extract a test list from a flutter drive log file. It was somewhat tested but not fully

**Comment by spauldhaliwal** [comment]
If we need to run a test with keyboard events, we are advised to use integration_test over flutter_driver as flutter_driver [doesn't support keyboard events](https://github.com/flutter/flutter/issues/9383) (such as arrow up, down, etc)

However, if we are testing performance, we should be testing in profile mode, which is unsupported by integration_test, and we are advised to use flutter_driver :/

Is there any workaround to running integration_test with profile or release modes? How are we expected to test performance if we can only run in debug mode, and according to flutter docs, integration_test is meant to replace flutter_driver?

**Comment by dallas1287** [comment]
Is there any update on this as an option. I think @spauldhaliwal put it pretty well that we're in a catch 22 here when we want to test performance or at the very least remove the overhead of the debugger especially when, in my companies case, we rely on multiple isolates, and are trying to stream multiple video streams simultaneously and want to be able to track our stream decoding without the latency caused by carrying a debugger with us.
